Babylon/groups/api/__init__.py

```diff
@@ -1,27 +1,30 @@
+import sys
 import cosmotech_api
-from azure.identity import AzureCliCredential
-from azure.identity import DefaultAzureCredential
-from click import group
-from click import pass_context
+from azure.identity import AzureCliCredential, DefaultAzureCredential
+from azure.core.exceptions import ClientAuthenticationError
+from click import group, pass_context
 from click.core import Context
 
 from .commands import list_commands
 from .groups import list_groups
 from ...utils.decorators import require_platform_key
 
-
 @group()
 @pass_context
 @require_platform_key("api_scope", "api_scope")
 @require_platform_key("api_url", "api_url")
 def api(ctx: Context, api_scope: str, api_url: str):
     """Group handling communication with the cosmotech API"""
-    try:
-        credentials = AzureCliCredential()
-    except:
-        credentials = DefaultAzureCredential()
+    token = None
+    for creds in [AzureCliCredential(), DefaultAzureCredential()]:
```

Comment:
We could only check with the `DefaultAzureCredential` since it uses all possible credentials the user could have set up. (the old version was a relic of the tests to get the `az cli` credentials to work and was not changed before release)